torchaudio/prototype/tacotron2.py

```diff
@@ -0,0 +1,851 @@
+                 decoder_no_early_stopping: bool = False) -> None:
+        super().__init__()
+
+        self.mask_padding = mask_padding
+        self.n_mels = n_mels
+        self.n_frames_per_step = n_frames_per_step
+        self.embedding = nn.Embedding(n_symbols, symbols_embedding_dim)
+        std = sqrt(2.0 / (n_symbols + symbols_embedding_dim))
+        val = sqrt(3.0) * std
+        self.embedding.weight.data.uniform_(-val, val)
+        self.encoder = _Encoder(encoder_n_convolutions,
+                                encoder_embedding_dim,
+                                encoder_kernel_size)
+        self.decoder = _Decoder(n_mels, n_frames_per_step,
+                                encoder_embedding_dim, attention_dim,
+                                attention_location_n_filters,
+                                attention_location_kernel_size,
+                                attention_rnn_dim, decoder_rnn_dim,
+                                prenet_dim, max_decoder_steps,
+                                gate_threshold, p_attention_dropout,
+                                p_decoder_dropout,
+                                not decoder_no_early_stopping)
+        self.postnet = _Postnet(n_mels, postnet_embedding_dim,
+                                postnet_kernel_size,
+                                postnet_n_convolutions)
+
+    def _parse_output(self,
+                      outputs: Tuple[Tensor, Tensor, Tensor, Tensor],
+                      output_lengths: Tensor) -> Tuple[Tensor, Tensor, Tensor, Tensor]:
+
+        if self.mask_padding and output_lengths is not None:
+            mask = get_mask_from_lengths(output_lengths)
+            mask = mask.expand(self.n_mels, mask.size(0), mask.size(1))
+            mask = mask.permute(1, 0, 2)
+
+            outputs[0].masked_fill_(mask, 0.0)
+            outputs[1].masked_fill_(mask, 0.0)
+            outputs[2].masked_fill_(mask[:, 0, :], 1e3)  # gate energies
+
+        return outputs
+
+    def forward(self,
+                text: Tensor,
+                text_lengths: Tensor,
+                mel_specgram: Tensor,
+                mel_specgram_lengths: Tensor) -> Tuple[Tensor, Tensor, Tensor, Tensor]:
+        r"""Pass the input through the Tacotron2 model. This is in teacher
+        forcing mode, which is generally used for training.
+
+        The input ``text`` should be padded with zeros to length text_lengths.max().
+        The input ``mel_specgram`` should be padded with zeros to length mel_specgram_lengths.max().
+
+        Args:
+            text (Tensor): The input text to Tacotron2 with shape (n_batch, text_lengths.max()).
+            text_lengths (Tensor): The length of each text with shape (n_batch).
+            mel_specgram (Tensor): The target mel spectrogram with shape (n_batch, n_mels, mel_specgram_lengths.max()).
+            mel_specgram_lengths (Tensor): The length of each mel spectrogram with shape (n_batch).
```

Comment:
```suggestion
            text (Tensor): The input text to Tacotron2 with shape (n_batch, max of text_lengths).
            text_lengths (Tensor): The length of each text with shape (n_batch).
            mel_specgram (Tensor): The target mel spectrogram with shape (n_batch, n_mel, max of mel_specgram_lengths).
            mel_specgram_lengths (Tensor): The length of each mel spectrogram with shape (n_batch).
```
or follow convention [here](https://github.com/pytorch/audio/pull/1620/commits/34cf5210ddfae17c89e1f013b261abb65b4b2e63) for instance. thoughts?

Reply:
see also [torchaudio naming convention](https://github.com/pytorch/audio/blob/master/README.md#conventions)